core/src/components/input/input.tsx

```diff
@@ -383,47 +385,48 @@ export class Input implements ComponentInterface {
 
   private onBlur = (ev: FocusEvent) => {
     this.hasFocus = false;
-    this.focusChanged();
     this.emitStyle();
 
-    if (this.inputCleared) {
-      if (this.focusedValue !== this.value) {
-        /**
-         * Emits the `ionChange` event when the input value
-         * is different than the value when the input was focused.
-         */
-        this.emitValueChange();
-      }
-      this.focusedValue = undefined;
-      this.inputCleared = false;
+    if (this.focusedValue !== this.value) {
+      /**
+       * Emits the `ionChange` event when the input value
+       * is different than the value when the input was focused.
+       */
+      this.emitValueChange();
     }
 
+    this.didInputClearOnEdit = false;
+
     this.ionBlur.emit(ev);
   };
 
   private onFocus = (ev: FocusEvent) => {
     this.hasFocus = true;
     this.focusedValue = this.value;
-    this.focusChanged();
     this.emitStyle();
 
     this.ionFocus.emit(ev);
   };
 
   private onKeydown = (ev: KeyboardEvent) => {
-    if (this.shouldClearOnEdit()) {
-      // Did the input value change after it was blurred and edited?
-      // Do not clear if user is hitting Enter to submit form
-      if (this.didBlurAfterEdit && this.hasValue() && ev.key !== 'Enter') {
-        // Clear the input
-        this.clearTextInput();
-      }
-
-      // Reset the flag
-      this.didBlurAfterEdit = false;
-    }
+    this.checkClearOnEdit(ev);
   };
 
+  private checkClearOnEdit(ev: KeyboardEvent) {
+    if (!this.shouldClearOnEdit()) {
+      return;
+    }
+    /**
+     * Clear the input if the control has not been previously cleared during focus.
+     * Do not clear if the user hitting enter to submit a form.
+     */
+    if (!this.didInputClearOnEdit && this.hasValue() && ev.key !== 'Enter') {
+      this.value = '';
+    }
+    this.didInputClearOnEdit = true;
+    this.ionInput.emit(null);
```

Comment:
This event doesn't seem to be getting emitted, but I'm not sure why it's needed anyway? In theory, wouldn't this cause duplicate `ionInput` emissions on clear -- one for the clear itself, and one for the key press that triggered the clear?

Recording of me typing while logging `ionInput` events (note only one event firing when input is cleared): https://user-images.githubusercontent.com/90629384/193068697-468f0cd0-c5c7-463b-89e9-c441f6a026ac.mp4

Edit: On closer examination, I _am_ seeing an `ionInput` event with `null` fire when pressing keys like Ctrl and Shift, which I'm not sure is intended.